open_diloco/train_pure_fsdp.py

```diff
@@ -0,0 +1,273 @@
+import os
+from contextlib import nullcontext
+import datetime
+from typing import Literal
+
+import torch
+import torch.distributed as dist
+from pydantic_config import parse_argv, BaseConfig
+from torch.distributed import destroy_process_group, init_process_group
+
+from torchdata.stateful_dataloader import StatefulDataLoader
+from transformers import (
+    AutoTokenizer,
+    DataCollatorForLanguageModeling,
+    LlamaConfig,
+    LlamaForCausalLM,
+    get_cosine_schedule_with_warmup,
+)
+from torch.distributed.fsdp import (
+    FullyShardedDataParallel as FSDP,
+    MixedPrecision,
+)
+from torch.distributed.device_mesh import init_device_mesh
+from hivemind.optim.optimizer import logger
```

Comment:
I think we might not need this logger from hivemind?

Reply:
we do, otherwise we don;t have a nice logger